src/main/webapp/app/components/sidebar/GenericSidebar.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import './generic-sidebar.scss';
+import { observer } from 'mobx-react';
+import { IRootStore } from 'app/stores';
+import { componentInject } from 'app/shared/util/typed-inject';
+import { FaChevronRight, FaChevronLeft } from 'react-icons/fa';
+
+export interface IGenericSidebarProps extends StoreProps {
+  children: React.ReactNode;
+}
+
+const GenericSidebar = ({ children, showGenericSidebar, toggleGenericSidebar, genericSidebarWidth }: IGenericSidebarProps) => {
+  return showGenericSidebar ? (
+    <div style={{ width: genericSidebarWidth }} className="generic-sidebar-expanded">
+      <div style={{ marginTop: '2rem', marginLeft: '1rem', display: 'flex' }}>
+        <FaChevronRight cursor={'pointer'} style={{ marginTop: '6px' }} onClick={() => toggleGenericSidebar(false)} />
+        {children}
+      </div>
+    </div>
+  ) : (
+    <div className="generic-sidebar-collapsed">
+      <FaChevronLeft
+        cursor={'pointer'}
+        style={{ position: 'absolute', right: '38', top: '38' }}
```

Comment:
What's this magic 38 number?

Reply:
It's just a position I chose that makes the chevron that opens the sidebar look better. It also aligns it with the y position of the chevron that closes the sidebar, so the transition is smooth. Should we make it a constant?